Makefile

```diff
@@ -1,32 +1,60 @@
 SHELL = /bin/bash
 
-config:
-	./scripts/ninja.js config
-
-build: config
-	./scripts/ninja.js build
+build:
 	dune build
+	./scripts/copyExes.js
 
-watch: config
+watch:
 	dune build -w
 
-dce: build
+bench:
+	dune exec -- syntax_benchmarks
+
+dce:
 	opam exec reanalyze.exe -- -dce-cmt _build
 
-test: build
-	npm test
+ninja/ninja:
+	./scripts/buildNinjaBinary.js
+
+ninja: ninja/ninja
+
+test: lib
+	dune exec -- node scripts/ciTest.js -all
+
+test-syntax: build
+	dune exec -- syntax_tests
+	dune exec -- bash ./scripts/test_syntax.sh
+	make reanalyze
+	bash ./scripts/testok.sh
+
+test-syntax-roundtrip: build
+	dune exec -- syntax_tests
+	ROUNDTRIP_TEST=1 dune exec -- bash ./scripts/test.sh
+	make reanalyze
+	bash ./scripts/testok.sh
 
 test-gentype: build
 	make -C jscomp/gentype_tests/typescript-react-example test
 
 test-all: test test-gentype
 
-lib: build
-	node scripts/install -force-lib-rebuild
+reanalyze:
+	reanalyze.exe -set-exit-code -all-cmt _build/default/res_syntax -suppress res_syntax/testrunner,res_syntax/compiler-libs-406 -exclude-paths res_syntax/compiler-libs-406
+
+lib: build ninja/ninja
+	dune exec -- node scripts/ninja.js config
+	dune exec -- node scripts/ninja.js build
 
 artifacts: lib
+	./scripts/prebuilt.js
 	./scripts/makeArtifactList.js
 
+format:
+	dune build @fmt --auto-promote
+
```

Comment:
do you want to promote files from `ml` directory which came from upstream?

Reply:
Currently, while there is a global .ocamlformat file, ocamlformat is disabled for the whole jscomp dir via jscomp/.ocamlformat. We can enable it selectively for the different jscomp subdirs in subsequent PRs.